deps/src/caller.cpp

```diff
@@ -0,0 +1,340 @@
+    for (int i = 0; i < size; i++) {
+        result_content[i] =
+            static_cast<int>(jl_unbox_int64(jl_arrayref(array, i)));
+    }
+    return result;
+}
+
+intvec* jl_array_to_intmat(jl_value_t* array_val){
+    jl_array_t * array = reinterpret_cast<jl_array_t *>(array_val);
+    int cols = jl_array_dim(array,0);
+    int rows = jl_array_dim(array,1);
+    intvec* result = new intvec(rows,cols);
+    int64_t* array_data = reinterpret_cast<int64_t*>(jl_array_data(array));
+    for(int i = 0; i < rows; i ++){
+        for( int j = 0; i < cols; i++){
+            IMATELEM(*result,i,j) = static_cast<int>(array_data[ j + (i*cols)]);
+        }
+    }
+    return result;
+}
+
+bool translate_singular_type(
+    jl_value_t * obj, void ** args, int * argtypes, int i, ring r)
+{
+    if (jl_isa(obj, jl_singular_number_type)) {
+        argtypes[i] = NUMBER_CMD;
+        args[i] = reinterpret_cast<void *>(n_Copy(
+            reinterpret_cast<number>(get_ptr_from_cxxwrap_obj(obj)), r->cf));
+        return true;
+    }
+    if (jl_isa(obj, jl_singular_ring_type)) {
+        argtypes[i] = RING_CMD;
+        void * rng = get_ptr_from_cxxwrap_obj(obj);
+        reinterpret_cast<ring>(rng)->ref++;
+        args[i] = rng;
+        return true;
+    }
+    if (jl_isa(obj, jl_singular_poly_type)) {
+        argtypes[i] = POLY_CMD;
+        args[i] = reinterpret_cast<void *>(
+            p_Copy(reinterpret_cast<poly>(get_ptr_from_cxxwrap_obj(obj)), r));
+        return true;
+    }
+    if (jl_isa(obj, jl_singular_ideal_type)) {
+        argtypes[i] = IDEAL_CMD;
+        args[i] = reinterpret_cast<void *>(id_Copy(
+            reinterpret_cast<ideal>(get_ptr_from_cxxwrap_obj(obj)), r));
+        return true;
+    }
+    if (jl_isa(obj, jl_singular_matrix_type)) {
+        argtypes[i] = MATRIX_CMD;
+        args[i] = reinterpret_cast<void *>(mp_Copy(
+            reinterpret_cast<matrix>(get_ptr_from_cxxwrap_obj(obj)), r));
+        return true;
+    }
+    if (jl_isa(obj, jl_singular_bigint_type)){
+        argtypes[i] = BIGINT_CMD;
+        args[i] = reinterpret_cast<void*>(get_ptr_from_cxxwrap_obj(obj));
+    }
+    if (jl_isa(obj, jl_singular_map_type)){
+        argtypes[i] = MAP_CMD;
+        args[i] = reinterpret_cast<void*>(get_ptr_from_cxxwrap_obj(obj));
+    }
+    if (jl_isa(obj, jl_int64_vector_type)) {
+        argtypes[i] = INTVEC_CMD;
+        args[i] = reinterpret_cast<void *>(jl_array_to_intvec(obj));
+        return true;
+    }
+    if (jl_isa(obj, jl_int64_matrix_type)) {
+        argtypes[i] = INTMAT_CMD;
+        args[i] = reinterpret_cast<void *>(jl_array_to_intmat(obj));
+        return true;
+    }
+    if (jl_isa(obj, jl_singular_bigintmat_type)) {
+        argtypes[i] = BIGINTMAT_CMD;
+        bigintmat * copy = new bigintmat(reinterpret_cast<bigintmat*>(get_ptr_from_cxxwrap_obj(obj)));
+        args[i] = reinterpret_cast<void*>(copy);
+        return true;
+    }
+    if (jl_isa(obj,jl_singular_resolution_type)) {
```

Comment:
```suggestion
    if (jl_isa(obj, jl_singular_resolution_type)) {
```

Reply:
I already removed this whole function . . .